osquery/core/query.cpp

```diff
@@ -169,30 +171,56 @@ Status Query::addNewResults(QueryData current_qd,
 
 Status serializeRow(const Row& r,
                     const ColumnNames& cols,
+                    const ColumnTypes& colTypes,
                     JSON& doc,
                     rj::Value& obj) {
-  if (cols.empty()) {
-    for (const auto& i : r) {
-      doc.addRef(i.first, i.second, obj);
-    }
-  } else {
-    for (const auto& c : cols) {
-      auto i = r.find(c);
-      if (i != r.end()) {
-        doc.addRef(c, i->second, obj);
+  for (const auto& i : r) {
+    if (cols.empty() ||
```

Comment:
no cols.empty() needed

Reply:
Without it, would we not change the behavior that was expressed in lines 174-178 of the previous version
```
 if (cols.empty()) {
    for (const auto& i : r) {
      doc.addRef(i.first, i.second, obj);
    }
 } else {
```
?

Reply:
Sorry false alarm, I thought it && instead of the ||. ( Recently removed some of those)